src/classes/SRVC_ContactRelationshipAutoCreate.cls

```diff
@@ -0,0 +1,178 @@
+            SRVC_ContactRelationshipAutoCreate.instance = new SRVC_ContactRelationshipAutoCreate();
+        }
+
+        return SRVC_ContactRelationshipAutoCreate.instance;
+    }
+
+    /**************************************************************************************************************************
+    * @description Retrieves a model representing the Relationship records to be created for a list of Contact records.
+    * @param contactsList A list of Contact records from which to retrieve associated Relationship records.
+    * @param contactRelationShipAutoCreateMapModelsList A List of MODL_ConRelationshipAutoCreateMapping representing Relationship Auto Create mappings for Contact.
+    * @return A MODL_ContactsRelationships model representing the Relationships associated with a list of Contacts. 
+    **************************************************************************************************************************/ 
+    public virtual MODL_ContactsRelationships getRelationshipsToCreateForContacts(List<Contact> contactsList, List<MODL_ConRelationshipAutoCreateMapping> contactRelationShipAutoCreateMapModelsList){
+        MODL_ContactsRelationships contactsRelationshipsModel = new MODL_ContactsRelationships();
+
+        for (Contact con : contactsList){
+            MODL_ContactRelationships conRelationshipsModel = this.getRelationshipsToCreateForContact(con, contactRelationShipAutoCreateMapModelsList);
+
+            if (conRelationshipsModel != null){
+                contactsRelationshipsModel.addContactRelationshipsModel(conRelationshipsModel);
+            }
+        }
+
+        return contactsRelationshipsModel;
+    }
+
+    /**************************************************************************************************************************
+    * @description Retrieves a model representing the Relationship records to be created for a single Contact record.
+    * @param  contact The Contact record to be used to create the ContactRelationships model.
+    * @param  contactRelationShipAutoCreateMapModelsList A List of MODL_ConRelationshipAutoCreateMapping representing Relationship Auto Create mappings for Contact.
+    * @return A MODL_ContactRelationships model representing the Relationships associated with the specified Contact record. 
+    **************************************************************************************************************************/ 
+    @TestVisible
+    protected virtual MODL_ContactRelationships getRelationshipsToCreateForContact(Contact contact, List<MODL_ConRelationshipAutoCreateMapping> contactRelationShipAutoCreateMapModelsList){
+        MODL_ContactRelationships conRelationshipsModel; 
+
+        for (MODL_ConRelationshipAutoCreateMapping conRelAutoCreateMap : contactRelationShipAutoCreateMapModelsList){
+            Object contactFieldValue = contact.get(conRelAutoCreateMap.contactFieldApiName);
+
+            if (contactFieldValue != null){
```

Comment:
@MatthewBlanski should we throw an error here if the contactFieldValue is not an Id?  The current functionality throws an error for invalid auto create mappings, but it looks like it gets removed in the refactor.  Would it make sense to do something like this:

try{
                    Relationship__c conRelationshipToAdd = new Relationship__c(Contact__c = contact.Id,
                                                                               RelatedContact__c = (Id)contactFieldValue,
                                                                               Type__c = conRelAutoCreateMap.relationshipType,
                                                                               Status__c = 'Current');
    
                    conRelationshipsModel = new MODL_ContactRelationships(contact.Id);
                    conRelationshipsModel.addRelationship(conRelationshipToAdd);
                } catch (Exception e){
                    contact.addError(system.Label.AutoCreateFieldError);
                }

Reply:
Change line 93 to an Id and cast contact.get() to an Id. That'll produce the Id mismatch error. Wrap that line only in the try catch block. Also refer to the Affiliations PR for the exceptions to catch for. Anything other than that we don't want to catch for.